### PR TITLE
don't hit sig/get if we don't have to

### PR DESCRIPTION
**Pull request by maxtaco**
- the merkle tree might say our cached copy is up-to-date, and we would have been hitting the sig/get endpoint just to get a nil update, so eliminate the RT

### CODE REVIEW

**Comment by maxtaco** [comment]
cc @mmaxim

File: go/teams/ftl.go [hunk 1]
```diff
@@ -1102,16 +1121,22 @@ func makeState(arg fastLoadArg, s *keybase1.FastTeamData) *keybase1.FastTeamData
...
 
-	groceries, err := f.loadFromServerWithRetries(m, arg, shoppingList)
+	groceries, err := f.loadFromServerWithRetries(m, arg, state, shoppingList)
 	if err != nil {
 		return nil, err
 	}
 
+	if groceries == nil {
+		m.CDebugf("FastTeamChainLoader#refresh: our state was fresh according to the Merkle tree")
+		return nil, nil
```
Comment (mlsteele) [inline go/teams/ftl.go]: why not return `state, nil`?

Comment (maxtaco) [reply go/teams/ftl.go]: So the result isn't re-cached.

Comment (mlsteele) [reply go/teams/ftl.go]: wicked fast

Comment (maxtaco) [reply go/teams/ftl.go]: I used to be threading through a bool to reflect freshness, but this seemed a little cleaner

**Pull request merged**